File: backend/src/models/fincaModel.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Finca = sequelize.define('Finca', {

  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true
  },

  nombre: {
    type: DataTypes.STRING,
    allowNull: false
  },

  departamento: {
    type: DataTypes.STRING,
    allowNull: false
  },

  municipio: {
    type: DataTypes.STRING,
    allowNull: false
  },

  area_total_hectareas: {
    type: DataTypes.FLOAT
  },

  latitud: {
    type: DataTypes.FLOAT
  },

  longitud: {
    type: DataTypes.FLOAT
  },

  fecha_registro: {
    type: DataTypes.DATE,
    allowNull: false,
    defaultValue: DataTypes.NOW
  }

}, {
  tableName: 'fincas',
  timestamps: false
});

module.exports = Finca;

File: mobile-app/lib/models/finca_model.dart
class Finca {

  String id;

  String nombre;

  String departamento;

  String municipio;

  double areaTotalHectareas;

  double latitud;

  double longitud;

  bool sincronizado;

  Finca({
    required this.id,
    required this.nombre,
    required this.departamento,
    required this.municipio,
    required this.areaTotalHectareas,
    required this.latitud,
    required this.longitud,
    required this.sincronizado,
  });

  Map<String, dynamic> toMap() {

    return {
      'id': id,
      'nombre': nombre,
      'departamento': departamento,
      'municipio': municipio,
      'area_total_hectareas': areaTotalHectareas,
      'latitud': latitud,
      'longitud': longitud,
      'sincronizado': sincronizado ? 1 : 0
    };

  }

  factory Finca.fromMap(Map<String, dynamic> map) {

    return Finca(
      id: map['id'],
      nombre: map['nombre'],
      departamento: map['departamento'],
      municipio: map['municipio'],
      areaTotalHectareas: map['area_total_hectareas'],
      latitud: map['latitud'],
      longitud: map['longitud'],
      sincronizado: map['sincronizado'] == 1,
    );

  }

}